linux/shim/service.go

```diff
@@ -30,14 +30,16 @@ func New(path string) *Service {
 }
 
 type Service struct {
-	initProcess *initProcess
-	path        string
-	id          string
-	bundle      string
-	mu          sync.Mutex
-	processes   map[int]process
-	events      chan *task.Event
-	execID      int
+	initProcess   *initProcess
+	path          string
+	id            string
+	bundle        string
+	mu            sync.Mutex
+	processes     map[int]process
+	events        chan *task.Event
+	eventsMu      sync.Mutex
+	deferredEvent *task.Event
```

Comment:
There can only ever be one by construction, the code always makes sure it sends the one outstanding event before it starts processing `s.events` again.